### PR TITLE
Added support for V3

### PR DESCRIPTION
**Pull request by ASH3xC0d3**
Added support for version 3.
All the existing issues with v1 and v2c are also with v3 e.g.
https://github.com/node-red/node-red-nodes/issues/679
snmp walker will not return more than 1000 objects, all following objects cannot be accessed
Incorrect encryption (privacy) passphrase gives a timeout and not an error (protocol limitation?)
Incorrect community string gives a timeout and not an error (protocol limitation?)
Changes to the node are made with following details:
Node-red v1.2.2
Node.js v12.6.0
Windows_NT 10.0.19841 x64 LE

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [contribution guidelines](https://github.com/node-red/node-red-nodes/blob/master/CONTRIBUTING.md)
- [x] For non-bugfix PRs, I have discussed this change on the forum/slack team.
- [ ] I have run `grunt` to verify the unit tests pass
- [x] I have added suitable unit tests to cover the new/changed functionality

### CODE REVIEW

File: io/snmp/snmp.html [hunk 1]
```diff
@@ -3,19 +3,56 @@
...
+            <option value="SHA">SHA</option>
+        </select>
+        <span style="margin-left:50px;">Priv.Prot.</span>
+        <select type="text" id="node-input-privprot" style="width:150px;">
+            <option value="None">None</option>
+            <option value="DES">DES</option>
+            <option value="AES">AES</option>
+        </select>
+    </div>
+    <div class="form-row">
```
Comment (dceejay) [inline io/snmp/snmp.html:47]: Likewise - do these need to be shown if "None" is selected above ?

File: io/snmp/snmp.html [hunk 2]
```diff
@@ -3,19 +3,56 @@
...
+        <label for="node-input-authkey"><i class="fa fa-user"></i> Auth.Key</label>
+        <input type="password" id="node-input-authkey" placeholder="Authentication key">
+    </div>
+    <div class="form-row">
```
Comment (dceejay) [inline io/snmp/snmp.html:51]: Same again

File: io/snmp/snmp.html [hunk 3]
```diff
@@ -42,10 +82,16 @@
...
             host: { value: "127.0.0.1" },
+            version: { value: "v1", required: true },
+            timeout: { value: 5 },
             community: { value: "public" },
-            version: { value: "1", required: true },
+            username: { value: "" },
+            auth: { value: "noAuthNoPriv", required: true },
+            authprot: { value: "None", required: true },
+            privprot: { value: "None", required: true },
+            authkey: { value: "" },
```
Comment (dceejay) [inline io/snmp/snmp.html:92]: Any credentials need to be handled as credentials and not general variables - see https://nodered.org/docs/creating-nodes/credentials (ie add a credential section like the defaults one but for the items that should be secret.)

**Review by dceejay — CHANGES_REQUESTED** [review]
So yes - two main things - 
1) is that we should hide any fields from the user that aren't necessary - so only show v3 relevant fields when v3 is selected. and keys when encryption is selected etc... A simple example to follow is in the smooth node 17-smooth.html line 35 and then 91, 96 . 
2) All credentials MUST be stored as credentials for security - so need to be handles as per docs - https://nodered.org/docs/creating-nodes/credentials - or for an example see the mysql node 68-mysql.html line 48 and 68-mysql.js line 39 and 97.

**Comment by Steve-Mcl** [comment]
@ASH3xC0d3 your changes have been incorporated into an updated PR #904 and merged into master

Thanks for the PR 👍

**Pull request closed without merging**